### PR TITLE
feat: Clonable project

### PR DESCRIPTION
**Pull request by kof**
## Description

Every project is now clonable if you have a shared link with at least view permission or higher.

## Steps for reproduction

1. get a link with view permission from another owner
2. clone it
3. repeat the same from your own project

## Code Review

- [ ] hi @istarkov , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/sdk/src/schema/pages.ts [hunk 1]
```diff
@@ -114,26 +115,24 @@ export const ProjectNewRedirectPathSchema = PagePath.or(
...
-export type ProjectMeta = z.infer<typeof ProjectMeta>;
+export type CompilerSettings = z.infer<typeof CompilerSettings>;
 
 export type Page = z.infer<typeof Page>;
 
 export const Pages = z.object({
   meta: ProjectMeta.optional(),
-  settings: ProjectSettings.optional(),
+  compiler: CompilerSettings.optional(),
+  redirects: z.array(PageRedirect).optional(),
```
Comment (TrySound) [inline packages/sdk/src/schema/pages.ts:135]: These seem like completely unrelated changes and will break atomic styles. Let's do renamings separately.

File: apps/builder/app/shared/router-utils/path-utils.ts [hunk 2]
```diff
@@ -73,6 +73,8 @@ export const builderDomainsPath = (method: string) => {
   }`;
 };
 
+export const projectPath = (method: string) => `/project/${method}`;
```
Comment (TrySound) [inline apps/builder/app/shared/router-utils/path-utils.ts:76]: doesn't constrain anything, maybe just write path where it is used?

File: apps/builder/app/shared/clone-project/clone-project.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import { useState } from "react";
```
Comment (TrySound) [inline apps/builder/app/shared/clone-project/clone-project.tsx:1]: apps/builder/app/shared/clone-project.tsx

File: apps/builder/app/builder/features/project-settings/meta-section.tsx [hunk 4]
```diff
@@ -26,29 +28,41 @@ const imgStyle = css({
...
-      });
-    };
+const defaultMetaSettings: ProjectMeta = {
+  siteName: "",
+  faviconAssetId: "",
+  code: "",
+};
 
+export const MetaSection = () => {
+  const [meta, setMeta] = useState($pages.get()?.meta ?? defaultMetaSettings);
```
Comment (TrySound) [inline apps/builder/app/builder/features/project-settings/meta-section.tsx:38]: please use function initializer, constant calling of .get in render deoptimizes stores

File: packages/sdk/src/schema/pages.ts (hunk 1 above)
Comment (kof) [reply packages/sdk/src/schema/pages.ts:135]: I needed to clean up settings, could be in stages, but really would take much more time to make in separate prs. I think nobody uses that css setting, but to be sure we can check in prod and add a migration if needed

File: apps/builder/app/shared/router-utils/path-utils.ts (hunk 2 above)
Comment (kof) [reply apps/builder/app/shared/router-utils/path-utils.ts:76]: we have all paths in this file, is spreading them across the codebase a good idea?

File: apps/builder/app/builder/features/project-settings/meta-section.tsx (hunk 4 above)
Comment (kof) [reply apps/builder/app/builder/features/project-settings/meta-section.tsx:38]: do you mean using useStore?

File: apps/builder/app/shared/router-utils/path-utils.ts (hunk 2 above)
Comment (TrySound) [reply apps/builder/app/shared/router-utils/path-utils.ts:76]: What the different slightly refined path would make? "method" would spread the same way

File: apps/builder/app/builder/features/project-settings/meta-section.tsx (hunk 4 above)
Comment (TrySound) [reply apps/builder/app/builder/features/project-settings/meta-section.tsx:38]: I mean useState(() => here)

Comment (kof) [reply apps/builder/app/builder/features/project-settings/meta-section.tsx:38]: actually since I changed how we save, we do it now right away without waiting for blur or close, we can as well use useStore, because there is no need to have a stale/static state any more, wdyt?

File: apps/builder/app/shared/router-utils/path-utils.ts (hunk 2 above)
Comment (kof) [reply apps/builder/app/shared/router-utils/path-utils.ts:76]: yes, its just about keeping all paths in one place and if needed extend from here in case we need a change

File: packages/project/src/shared/schema.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ export const Title = z
 
 export const ClonableSettings = z.object({
   isClonable: z.boolean().optional(),
+  isPublic: z.boolean().optional(),
```
Comment (TrySound) [inline packages/project/src/shared/schema.ts:14]: Aren't those exclusive?

Comment (kof) [reply packages/project/src/shared/schema.ts:14]: no, public is meant as in can be listed publicly on a directory, where clonable can be any project with even view link permission

Comment (TrySound) [reply packages/project/src/shared/schema.ts:14]: I'd replace isPublic with  visibility: private | public
